monai/handlers/checkpoint_loader.py

```diff
@@ -79,11 +84,36 @@ def __call__(self, engine: Engine) -> None:
         Args:
             engine: Ignite Engine, it can be a trainer, validator or evaluator.
         """
-        checkpoint = torch.load(self.load_path, map_location=self.map_location)
-        if len(self.load_dict) == 1:
-            key = list(self.load_dict.keys())[0]
-            if not (key in checkpoint):
-                checkpoint = {key: checkpoint}
+        if self.find_latest:
+            models = sorted(os.listdir(self.load_path))
+            self.logger.info(
+                "This directory already exisits, looking for previous models.")
+            if len(models) > 1:
+                list_model_time = []
+                times = []
+                for model_ in models:
+                    loc = os.path.join(self.load_path, model_)
+                    edit_time = os.path.getmtime(loc)
+                    times.append(edit_time)
+                    list_model_time.append([edit_time, loc])
+                list_model_time.sort(key=lambda x: x[0])
```

Comment:
You can extract the most recent model file with `max((os.path.join(self.load_path, m) for m in models), os.path.getmtime)`.